compiler/resolution/functionResolution.cpp

```diff
@@ -8326,15 +8322,24 @@ static void insertAutoCopyForArrayReturn(FnSymbol* fn) {
                 copiedRHS = toSymExpr(rhsCall->get(1)->copy());
                 call->get(2)->remove();
               } else {
-                VarSymbol* tmp = newTemp("array_autoCopy_ret_tmp", rhsType);
+                VarSymbol* tmp = newTemp("array_unref_ret_tmp", rhsType);
                 call->insertBefore(new DefExpr(tmp));
                 call->insertBefore(new CallExpr(PRIM_MOVE, tmp, call->get(2)->remove()));
                 copiedRHS = new SymExpr(tmp);
               }
             }
-            FnSymbol* autoFn = getAutoCopy(rhsType);
-            CallExpr* copy = new CallExpr(autoFn, copiedRHS);
-            call->insertAtTail(copy);
+
+            CallExpr* copy = new CallExpr("chpl__unref", copiedRHS);
+            FnSymbol* unrefFn = resolveUninsertedCall(NULL, call, copy, false);
```

Comment:
since resolveUninsertedCall just adds it to the AST and then removes it, I'd just add it yourself here and then call resolve on it, and remove it in the case you don't want to work on.